### PR TITLE
Team Name Recovery Page

### PR DESCRIPTION
**Pull request by RandomAshes**
## Overview
* This PR adds a team name recovery page that is linked to from the request password reset page.
* In this PR I also moved the endpoints that are sending emails into an EmailController as the team service was getting hard to follow which endpoints were responsible for teams and which were responsible for other things.

### Demo
Screen Shot 2022-10-07 at 12 08 01 PM

Screen Shot 2022-10-07 at 10 58 36 AM

Screen Shot 2022-10-07 at 12 01 15 PM

## Testing Instructions
1) Go to `http://localhost:3000/request-password-reset` and click the `I don't remember my team name` link.
2) Type in an incorrect email and submit. Ensure there is an error message saying that that email is not associated with any teams.
3) Type in an email address that is associated with at least one team. Submit form.
4) Confirm that there is a confirmation message saying to check your mail, what email address you just typed in, and what email address to expect the email to come from.
5) Check that an email has been sent (this part can only be tested in dev) and that it includes all team names that email is associated with.

### CODE REVIEW

File: api/src/main/java/com/ford/labs/retroquest/email/EmailController.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "/api/email")
+public class EmailController {
```
Comment (RandomAshes) [inline api/src/main/java/com/ford/labs/retroquest/email/EmailController.java:35]: The team controller is getting very cluttered and I feel like it makes more sense to have endpoints that are just sending emails be in their own controller. I'll move the other endpoints over soon.

**Review by jwcherney — APPROVED** [review]

**Review by jwcherney — APPROVED** [review]

**Review by jnsartwell — APPROVED** [review]
